pandas/io/common.py

```diff
@@ -1002,6 +1003,22 @@ def write_to_buffer(self) -> None:
         self.buffer.addfile(tarinfo, self)
 
 
+class _BZ2File(bz2.BZ2File):
```

Comment:
It would be nice to share the method between `_BZ2File` and `_LZMAFile` - could have a mixin class but that would probably not work with the super call.

Reply:
Yeah was thinking the same thing. May actually break it out as a utility function, which could also aid in testing